src/components/splat.js

```diff
@@ -0,0 +1,910 @@
+      this.covAndColorTexture.needsUpdate = true;
+
+      var quadGeometry = new THREE.BufferGeometry();
+      // Why 2.0?
+      var vertices = new Float32Array([
+        -2.0, -2.0, 0.0,
+        2.0, 2.0, 0.0,
+        -2.0, 2.0, 0.0,
+
+        2.0, -2.0, 0.0,
+        2.0, 2.0, 0.0,
+        -2.0, -2.0, 0.0
+      ]);
+      quadGeometry.setAttribute('position', new THREE.BufferAttribute(vertices, 3));
+
+      var splatIndexArray = new Uint32Array(this.bufferTextureWidth * this.bufferTextureHeight);
+      var splatIndices = new THREE.InstancedBufferAttribute(splatIndexArray, 1, false);
+      splatIndices.setUsage(THREE.DynamicDrawUsage);
+      var geometry = new THREE.InstancedBufferGeometry().copy(quadGeometry);
+      geometry.setAttribute('splatIndex', splatIndices);
+      geometry.instanceCount = 1;
+
+      var material = new THREE.ShaderMaterial({
+        uniforms: {
+          viewport: {value: new Float32Array([1980, 1080])}, // Dummy. will be overwritten
+          focal: {value: 1000.0}, // Dummy. will be overwritten
+          centerAndScaleTexture: {value: this.centerAndScaleTexture},
+          covAndColorTexture: {value: this.covAndColorTexture},
+          gsProjectionMatrix: {value: this.getProjectionMatrix()},
+          gsModelViewMatrix: {value: this.getModelViewMatrix()}
+        },
+        vertexShader: this.vertexShader,
+        fragmentShader: this.fragmentShader,
+        blending: THREE.CustomBlending,
+        blendSrcAlpha: THREE.OneFactor,
+        depthTest: true,
+        depthWrite: false,
+        transparent: true
+      });
+
+      material.onBeforeRender = (renderer, scene, camera, geometry, object, group) => {
+        var projectionMatrix = this.getProjectionMatrix(camera);
+        mesh.material.uniforms.gsProjectionMatrix.value = projectionMatrix;
+        mesh.material.uniforms.gsModelViewMatrix.value = this.getModelViewMatrix(camera);
+
+        var viewport = new THREE.Vector4();
+        renderer.getCurrentViewport(viewport);
+        var focal = (viewport.w / 2.0) * Math.abs(-1);
+        material.uniforms.viewport.value[0] = viewport.z;
+        material.uniforms.viewport.value[1] = viewport.w;
+        material.uniforms.focal.value = focal;
+      };
+
+      var mesh = this.mesh = new THREE.Mesh(geometry, material);
+      mesh.frustumCulled = false;
+      this.el.setObject3D('mesh', mesh);
+
+      // Wait until texture is ready
+      while (true) {
+        var centerAndScaleTextureProperties = renderer.properties.get(this.centerAndScaleTexture);
+        var covAndColorTextureProperties = renderer.properties.get(this.covAndColorTexture);
+        if (centerAndScaleTextureProperties && centerAndScaleTextureProperties.__webglTexture &&
+          covAndColorTextureProperties && centerAndScaleTextureProperties.__webglTexture) {
+          break;
+        }
+        await new Promise(resolve => setTimeout(resolve, 10));
+      }
```

Comment:
Instead of this polling it should just tell Three.js to initialize the textures: https://threejs.org/docs/index.html#api/en/renderers/WebGLRenderer.initTexture